homework2/reviewapp/src/components/features/shows/ReviewForm/ReviewForm.tsx

```diff
@@ -0,0 +1,45 @@
+import StarsRating from "@/components/shared/StarsRating/StarsRating";
+import StarIconComponent from "@/components/shared/StarsRating/StarsRating";
+import { IReview } from "@/typings/review";
+import { Box, Button, Flex, Input } from "@chakra-ui/react";
+import { useState } from "react";
+
+export interface IReviewFormProps
+{
+    onAddReview:(review:IReview)=>void;
+}
+export const ReviewForm=({onAddReview}:IReviewFormProps) =>
+    {
+        const [rating, setRating] = useState(0);
+
+       const handleRatingChange = (selectedRating: number) => {
+        setRating(selectedRating); };
+        const onClickHandler=()=>
+            {
+                const reviewInputComment= document.getElementById('comment-input') as HTMLInputElement;
+                const commentValue=reviewInputComment.value;
+                if( rating==0 || commentValue==='')
+                    return;
+                const newReview:IReview={
+                    comment: commentValue,
+                    rating: rating
+
+                };
+                onAddReview(newReview);
+                reviewInputComment.value="";
+                setRating(0);
+
+
+            };
+        return (
+            <Flex gap={3}  textColor="grey"   direction={'column'} >
```

Comment:
If we named our component `ReviewForm`, I would personally expect a form element here.
Then you could handle form onSubmit event and remove `onClick` from the button by adding type="submit" on it.
It is semantically more correct and something we will do in future lessons by default :)

Reply:
This is fixed